dask/array/fft.py

```diff
@@ -221,42 +218,33 @@ def func(a, n=None, axis=None):
 ihfft = fft_wrap(np.fft.ihfft, dtype=np.complex_)
 
 
-def _fftfreq_helper(n, d=1.0, chunks=None, real=False):
+@wraps(np.fft.fftfreq)
+def fftfreq(n, d=1.0, chunks=None):
     n = int(n)
+    d = float(d)
 
-    l = n + 1
-    s = n // 2 + 1 if real else n
-    t = l - s
+    def _fftfreq_block(i, n, d):
+        r = i.astype(float)
+        r[i >= (n + 1) // 2] -= n
+        r /= n * d
+        return r
```

Comment:
Mapped functions should be defined in the top-level namespace so that they serialize well. This becomes important in distributed settings.